enhancements/oc/cli-manager.md

```diff
@@ -0,0 +1,334 @@
+---
+title: cli-manager
+authors:
+  - "@sallyom"
+  - "@deejross"
+reviewers:
+  - "@soltysh"
+  - "@jwmatthews"
+approvers:
+  - "@soltysh"
+  - "@sferich888"
+  - "@deads2k"
+  - "@spadgett"
+api-approvers:
+  - "@deads2k"
+  - "@soltysh"
+creation-date: 2021-10-06
+last-updated: 2021-10-18
+tracking-link:
+  - https://issues.redhat.com/browse/WRKLDS-169
+see-also:
+  - "/enhancements/installer/component-selection.md"
+---
+
+# OpenShift CLI Manager
+
+## Release Signoff Checklist
+
+- [X] Enhancement is `implementable`
+- [X] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+This proposal is describing the mechanism for how authors of a Command Line Interface (CLI) tool such as `oc`, `kubectl`, `odo`, `istio`, `tekton`, or `knative`,
+can deliver tools to OpenShift clusters in disconnected environments.  A feature is needed to manage various CLIs available for OpenShift and related services. The goal is for
+a connected user to discover, install, and upgrade tools that are compatible with the current cluster version easily and from a single location.
+
+`krew` is an upstream project to distribute CLI tools (plugins) to Kubernetes users today.
+It works by reading a Git repository of files describing the plugins,
+and providing download links to them for various different OS and architecture combinations.
+Since those download links in the default index are Internet-facing, a Git and file server would need to be
+setup by customers to create their own custom index for use in disconnected environments.
+
+In order to avoid creating a new protocol and tool for this functionality, this proposal aims to leverage the `krew` project and a new custom index feature provided by `krew` in the form of a new Controller. The index will be managed by cluster-admins using
+[Custom Resources](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/) (or CRs). An image registry will be used to host images that contain the binaries.
+
+By leveraging an image registry, which is an existing dependency for OpenShift, and allowing the index to be managed
```

Comment:
you say this as if it means the feature is dependent on the OCP internal registry, but i see nothing in the design that requires any images(even for our own plugins) be hosted in the internal registry (i'd expect to get our plugin images from registry.redhat.io, not via/through the internal registry).

in fact i think it is best that this feature *not* take a dependency on the internal registry, as we may want to make that component more optional in the future.

Reply:
This is mostly saying that you need an image registry to run OpenShift in disconnected environments already, so you won't need to stand up additional infrastructure (unless you want to) to support this functionality.

Reply:
got it.